src/runtime/prefetch.cpp

```diff
@@ -0,0 +1,13 @@
+#include "HalideRuntime.h"
+
+extern "C" {
+
+// These need to inline, otherwise the extern call with the ptr
+// parameter breaks a lot of optimizations.
+__attribute__((always_inline))
+WEAK int halide_prefetch(const void *ptr) {
```

Comment:
Naming it this way will cause tutorial_15 to fail; it assumes that any callable symbol beginning with "halide_" is part of the runtime, and will fail since this gets included in generated code even when no-runtime is specified.

Suggest you either rename this to have a different prefix, or avoid inlining it. (Sounds like the latter is not an option.)